client/finality-grandpa/src/communication/mod.rs

```diff
@@ -137,7 +136,11 @@ pub(crate) struct NetworkBridge<B: BlockT, N: Network<B>> {
 	neighbor_sender: periodic::NeighborPacketSender<B>,
 }
 
-impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
+impl<B: BlockT, N: Network<B>> NetworkBridge<B, N>
+	where
+		B::Hash: Unpin,
```

Comment:
I don't know why you need that, but it's most certainly wrong.
At no point do we pin the `Hash` or block number

Reply:
There's probably some struct that implements `Future` somewhere whose `Unpin` trait implementation is automatically derived from its parameter while it doesn't need to and could be derived unconditonally.

Reply:
That's probably a result of me using `Pin::into_inner(self)` when I don't need to. I'll try and sort that out.

Reply:
Ok, resolved